libshaderc/include/shaderc.hpp

```diff
@@ -109,21 +109,29 @@ class CompileOptions {
     other.options_ = nullptr;
   }
 
-  // Adds a predefined macro to the compilation options.
+  // Adds a predefined macro to the compilation options. The macro name and
+  // value are passed in with their char* pointers of their strings and their
+  // lengths. And the strings must remain valid for the duration of the call,
+  // but can be modified or deleted after this function has returned. The macro
+  // value argument can either be a nullptr or point to an empty string in case
+  // of adding a valueless macro. In case of valueless macro, the length of the
+  // value string will be ignored.
   // For further details refer to shaderc_compile_options_add_macro_definition
```

Comment:
In fact, all the details are (redundantly) spelled out above -- there are no "further details" in `shaderc.h` right now.